gbpservice/nfp/configurator/drivers/firewall/paloalto/paloalto_fw_driver.py

```diff
@@ -0,0 +1,875 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import ast
+import requests
+
+from neutron import context
+from oslo_log import log as logging
+from oslo_serialization import jsonutils
+
+from gbpservice.nfp.configurator.drivers.base import base_driver
+from gbpservice.nfp.configurator.lib import fw_constants as const
+
+import sys
+import json
+import pan.xapi
+import pan.commit
+import pan.config
+import time
+
+LOG = logging.getLogger(__name__)
```

Comment:
We need to fix the logging. This has been enhanced recently. Please check Vyos firewall for this.